unittests/test_cli.py

```diff
@@ -433,6 +434,21 @@ def test_show_env_config_unknown_env(self):
         assert 'Traceback' not in stderr
         assert 1 == returncode
 
+    def test_suppress_deprecation(self):
+        self.action = 'run'
+        self.checkpath = [
+            'unittests/resources/checks_unlisted/deprecated_test.py'
+        ]
+        with pytest.warns(ReframeDeprecationWarning):
+            returncode, stdout, stderr = self._run_reframe()
+
+        self.more_options = ['--suppress-deprecation']
+        with pytest.warns(None) as record:
+            self._run_reframe()
+
+        assert not any(isinstance(r.message, ReframeDeprecationWarning) for
+                       r in record)
```

Comment:
Where is the documentation for the return value of the `pytest.warns()`? I had to look into the code, but it's a bit hard to understand how a message can be of an exception type.

```suggestion
        assert not any(isinstance(w.message, ReframeDeprecationWarning)
                       for w in warnings)
```